widget/button.go

```diff
@@ -750,3 +794,11 @@ func (b *Button) createWidget() {
 
 	b.initText()
 }
+
+func (b *Button) onMask(x, y int) bool {
+	if b.MaskImage == nil {
+		return true
+	}
+	c := b.MaskImage.At(x, y)
```

Comment:
I like the idea, one concern is that .At is surprisingly non-performant:

Don't call (*Image).At too much
At is also heavy that tries to solve all the queued draw commands and retrieve pixels from GPU.

https://ebitengine.org/en/documents/performancetips.html

Reply:
I wonder if we dumped the pixels out func (i *[Image](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2#Image)) ReadPixels(pixels [][byte](https://pkg.go.dev/builtin#byte))func (i *[Image](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2#Image)) ReadPixels(pixels [][byte](https://pkg.go.dev/builtin#byte)) it would be more efficient

Reply:
Instead of using "(*Image).At" the idea is to store the pixels once in a byte array during SetLocation and afterwards reading from the byte array rather using .At correct?

Reply:
I changed the code to use readPixels and immediatly convert it into a 2d bool slice which holds simple true/false values. This will make the initialization step a bit slower, but the actual live evaluation a lot faster.